pkg/osutil/user.go

```diff
@@ -30,6 +30,16 @@ type Group struct {
 var users map[string]User
 var groups map[string]Group
 
+// regexUidGid detects valid Linux uid or gid.
+var regexUidGid = regexp.MustCompile("^[0-9]+$") //nolint:revive
+
+// regexUsername matches user and group names to be valid for `useradd`.
+// It allows a trailing '$', but it feels prudent to map those to the fallback user as well.
```

Comment:
I realize that this is mostly just a moved comment, but it feels ambiguous: the regex does **not** allow a trailing `$`, even though `useradd` does. So maybe it should be reworded something like

```suggestion
// `useradd` allows names with a trailing '$', but it feels prudent to map those
// names to the fallback user as well, so the regex does not allow them.
```